File: basic-java-app/app/src/main/java/com/driouechemed/basicjavaapp/database/dao/TaskDao.java
package com.driouechemed.basicjavaapp.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.driouechemed.basicjavaapp.database.entities.Task;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

@Dao
public interface TaskDao {

    @Insert
    Completable insertTask(Task task);

    @Query("SELECT * FROM task")
    Observable<List<Task>> getAllTasks();
}

File: basic-kotlin-app/app/src/main/java/com/driouechemed/basickotlinapp/database/dao/TaskDao.kt
package com.driouechemed.basickotlinapp.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.driouechemed.basickotlinapp.database.entities.Task
import kotlinx.coroutines.flow.Flow

@Dao
interface TaskDao {

    @Query("SELECT * FROM task")
    fun getAll(): Flow<List<Task>>

    @Insert
    suspend fun insert(task: Task)
}